test/testdata/resolver/optional_nil.rb.flatten-tree.exp

```diff
@@ -56,26 +56,26 @@ begin
         ::Sorbet::Private::Static.sig(<self>) do ||
           <self>.params({:"w" => ::String}).returns(::String)
         end
+        <self>.extend(::T::Sig)
+        ::Sorbet::Private::Static.keep_def(<self>, :"foo")
+        ::Sorbet::Private::Static.keep_def(<self>, :"bar")
+        ::Sorbet::Private::Static.keep_def(<self>, :"qux")
+        ::Sorbet::Private::Static.keep_def(<self>, :"baz")
```

Comment:
So I looked into what's going on here: I think that the re-ordering here shouldn't matter.

rewriter/Flatten.cc basically looks for all sigs and methods and moves them to the top of the class RHS, but keeps the other calls like `extend T::Sig` and friends in the same position relative to where they were before.

then DefaultArgs comes along and inserts new sigs, methods, and keep_def calls. but since it doesn't know where the sig+method defs stop and the "other stats" start, it doesn't try to insert the sig+method defs with the existing sig+method defs and the keep_defs with the existing "other stats".